idna/core.py

```diff
@@ -84,7 +88,7 @@ def check_bidi(label, check_ltr=False):
         raise IDNABidiError('First codepoint in label {} must be directionality L, R or AL'.format(repr(label)))
 
     valid_ending = False
-    number_type = False
+    number_type = False  # type: Union[str, bool]
```

Comment:
Could this be simplified into `number_type = None  # type: str` by also updating the condition on L106 to `if number_type is None`?

(`unicodedata.bidirectional` may return an empty string value)

If so, perhaps fair to do that in a separate changeset if it'd be misleading or confusing to apply it at the same time as the other changes here.

Reply:
Ah, perhaps `number_type = None  # type: Optional[str]` would be the way to go (making the nullability of the type explicit)

Reply:
Changed this to `Optional[str]`

Reply:
Thanks!